<a href="https://colab.research.google.com/github/NadezhdaGruzdeva/Statistics-Probability/blob/main/Statistics%26Probability_HW1.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

###Задача 1  
Из колоды в 52 карты извлекаются случайным образом 4 карты.  
a) Найти вероятность того, что все карты – крести.  
б) Найти вероятность, что среди 4-х карт окажется хотя бы один туз.

1.1. Найти вероятность того, что все карты – крести.  
Всего 52 карт из них:  
- 13 карт крести  
- 39 карт не крести  
P(A) = m/n  
P(A) - вероятность того, что все карты – крести  
m - число различных вариантов вытащить 4 крести  
n - число всех сочетаний 4 карт из 52 

1.1.1. Создадим функцию для расчета сочетаний

In [ ]:
import math

In [ ]:
def combination(k,n): 
  '''Функция для сочетний
  k - количество элементов в наборе
  n - общее число элементов
  '''
  return int(math.factorial(n) / (math.factorial(k) * math.factorial(n - k)))

**1.1**.2. Найдем m (число различных вариантов вытащить 4 крести) как количество сочений 4 элементов из множества 9 элементов

In [ ]:
m = combination(4,13)
m

715

1.1.3. Найдем n (число всех сочетаний 4 карт из 52) как количество сочений 4 элементов из множества 52 элементов

In [ ]:
n = combination(4,52)
n

270725

1.1*.4*. Найдем вероятность вытащить 4 крести

In [ ]:
P = m/n
P

0.0026410564225690276

Ответ: 0,264%

1.2. Найти вероятность, что среди 4-х карт окажется хотя бы один туз.  
Всего 36 карт из них:  
- 4 туза  
- 48 не туза  

*Пойдем по пути подсказанному вами и Демидом на семинаре, и найдем все варианты вытащить не туз, затем вычтем это число из общего числа вариантов, и получится число вариантов, где есть хотя бы 1 туз*  
P(A) = m/n  
P(A) - вероятность того, что есть туз в руке 
m - число различных вариантов вытащить хотя бы 1 туз
n - число всех сочетаний 4 карт из 52 

m = n - t  
t = число различных вариантов без тузов

In [ ]:
n = combination(4,52)
n

270725

In [ ]:
t = combination(4,48)
t

194580

In [ ]:
m = n - t
m

76145

In [ ]:
P = m/n
P

0.2812632745405855

Ответ: 28%

###Задача 2  
На входной двери подъезда установлен кодовый замок, содержащий десять кнопок с цифрами от 0 до 9.  
Код содержит три цифры, которые нужно нажать одновременно.  
Какова вероятность того, что человек, не знающий код, откроет дверь с первой попытки?

*Также используем сочетание для решение данной задачи, используя ранее созданную функцию*

P(A) = m/n
P(A) - вероятность того, что ввели правильный код  
m - число благоприятствующмх вариантов  
n - общее число вариантов

In [ ]:
'Вариант конечно очивидно 1 , но вот формальная проверка'
m = combination(3,3)
m

1

In [ ]:
n = combination(3,10)
n

120

In [ ]:
P = m/n
P

0.008333333333333333

Ответ: 0,83% или 1 к 120.

###Задача 3  
В ящике имеется 15 деталей, из которых 9 окрашены.  
Рабочий случайным образом извлекает 3 детали.  
Какова вероятность того, что все извлеченные детали окрашены?

*И снова сочетания)*  
P(A) = m/n  
P(A) - вероятность  
m - число благоприятствующмх вариантов  
n - общее число вариантов  

9 деталей окрашенны  
6 - не окрашены

In [ ]:
m = combination(3,9)
m

84

In [ ]:
n = combination(3,15)
n

455

In [ ]:
P = m/n
P

0.18461538461538463

Ответ: 18,5%

###Задача 4  
В лотерее 100 билетов. Из них 2 выигрышных.  
Какова вероятность того, что 2 приобретенных билета окажутся выигрышными?

Решение аналогично задаче 2

In [ ]:
n = combination(2,100)
n

4950

In [ ]:
P = 1/n
P

0.00020202020202020202

Ответ: 0.02% или 1 шанс из 4950

**Дополнительно для чата размещение**


In [ ]:
def razmeshenie(k,n): 
  '''Функция для размещения
  k - количество элементов в наборе
  n - общее число элементов
  '''
  return int(math.factorial(n) / math.factorial(n - k))

In [ ]:
n = razmeshenie(3,10)
n

720

In [ ]:
P= 1/n
P

0.001388888888888889